.github/actions/common_setup/action.yaml

```diff
@@ -0,0 +1,23 @@
+name: Common setup steps
+description: Install Ruby into the runner environment, install gem dependencies, and install Puppet modules
+runs:
+  using: composite
+  steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
```

Comment:
I don't think this action is available until you checkout the repo. Removing this step should allow us to use this as an action in the rest of the workflows for setting up Ruby/gems/modules, unless there's something I'm missing.